pkg/controller/actions/resourcecacher/resourcecacher.go

```diff
@@ -0,0 +1,71 @@
+package resourcecacher
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/apis/common"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/cacher"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/render"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+)
+
+type ResourceCacher struct {
+	cacher.Cacher
+}
+
+func (s *ResourceCacher) SetKeyFn(key cacher.CachingKeyFn) {
+	s.Cacher.SetKeyFn(key)
+}
+
+func (*ResourceCacher) Name() string {
+	return "ResourceCacher"
```

Comment:
since this ends up in a prometheus counter, it should probably be better to use a lowercase name

Reply:
The idea is to override it from the actual renderer (template/kustomize). Put it as a field?

Reply:
something like that yes, so we can distinguish between resource rendered by renderer